graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/Message.java

```diff
@@ -253,6 +253,9 @@ public void addField(final String key, final Object value) {
             if (LOG.isDebugEnabled()) {
                 LOG.debug("Ignoring invalid or reserved key {} for message {}", key, getId());
             }
+            if (key != null && key.contains(".")) {
+                LOG.warn("Ignoring field \"{}\"=\"{}\" - Keys with a \".\" character are not supported anymore since Graylog 2.0! (Message[{}])", key, value, toString());
```

Comment:
I would also the "since Graylog 2.0" part in the warning message. In the end users don't care since when a feature didn't work anymore. :wink: